app/feature/feature-home/src/main/kotlin/com/hedvig/android/feature/home/home/data/SeenImportantMessagesStorage.kt

```diff
@@ -1,20 +1,21 @@
 package com.hedvig.android.feature.home.home.data
 
+import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.update
 
 interface SeenImportantMessagesStorage {
-  fun hasSeenMessage(id: String): Boolean
+
+  val seenMessages: Flow<List<String>>
 
   fun markMessageAsSeen(id: String)
 }
 
 internal class SeenImportantMessagesStorageImpl : SeenImportantMessagesStorage {
   private val storedSeenMessages: MutableStateFlow<List<String>> = MutableStateFlow(listOf())
 
-  override fun hasSeenMessage(id: String): Boolean {
-    return storedSeenMessages.value.contains(id)
-  }
+  override val seenMessages: Flow<List<String>>
+    get() = storedSeenMessages
```

Comment:
Oh and now that it's in this shape, you could even turn this into a StateFlow, since it's backed by a MutableStateFlow anyway. And change it in the interface too.

```
override val seenMessages: StateFlow<List<String>> = storedSeenMessages.asStateFlow()
```

I think would work fine here